### DISCUSSION TITLE
Support capacitor.config.js/capacitor.config.ts

### DISCUSSION THREAD

**Discussion by razorness** [quasarframework/quasar#16121] [Ideas / Proposals] [answered]
It would be nice to support `capacitor.config.ts` for extended options and implicit type checks by Typescript.

Example:
```ts
import { CapacitorConfig } from '@capacitor/cli';

const config: CapacitorConfig = {
  appId: 'com.company.appname',
  appName: 'My Capacitor App',
  webDir: 'www',
};

export default config;
```

**Comment by yusufkandemir** [comment]
In case anyone wants to contribute, here are the places where the config file gets read:
https://github.com/quasarframework/quasar/blob/0cd5f34dd1b206976b8776f790a4c03cae4360f2/app-vite/lib/modes/capacitor/config-file.js#L70-L73
https://github.com/quasarframework/quasar/blob/0cd5f34dd1b206976b8776f790a4c03cae4360f2/app-webpack/lib/modes/capacitor/config-file.js#L70-L73
Unless you introduce huge, problematic, not well-maintained, etc. libraries, we can accept PRs.

**Comment by maximilianfixl** [comment]
In my case, dynamic intervention in the config would also be an advantage. I am currently looking for a solution to insert the address of the Vite server `.env` controlled.

```
"server": {
    "allowNavigation": [
      "auth.my-auth-domain.com"
    ],
    "androidScheme": "https",
    "iosScheme": "https",
    "url": "https://10.1.0.54:9500",
    "cleartext": true
  },
```

The background is that we do not want to add the capacitor.config.json to `.gitignore` in this project and the IP is different after each local build (last commit on a different machine).

Do you have any advice on how to solve this with the JSON format? How do you work at this point in Teams and git?

**Comment by yusufkandemir** [comment, marked as answer]
In app-vite v3 we added support for this:
- [feat(app-vite): capacitor.config.ts/.js support + env-driven runtime](https://github.com/quasarframework/quasar/commit/ff446d8e25fb26f1ad2e5a2b41f85d04ed9723d7)

You can also access Quasar CLI's built-in env variables and user-defined env variables in there. So, it also solves @maximilianfixl's use case.
